### PR TITLE
Parametricmeshloader

### PR DESCRIPTION
**Pull request by sescaida**
Trying to improve the parametric mesh loader:
- Refactoring so it accepts surface _and_ volumetric mesh generation
- Some other minor refactoring (naming)
- Including an example of a parametric mesh loader template to STLIB

### CODE REVIEW

**Comment by sescaida** [comment]
An additional comment: This feature is still experimental and I think it could be subject to change in the future. For example: the hashes are now a part of the filename, which delivers an elegant solution in the code, but could potentially be confusing, because with repeated generation there will be more and more files that the user can only difficultly tell apart. Therefore, there could be more than change before a more stable version.

**Comment by marques-bruno** [comment]
[ci-build]

File: python/stlib/physics/deformable/elasticmaterialobject.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def __init__(self,
                                            volumeMeshFileName, name, rotation, translation, scale, surfaceMeshFileName,
                                            collisionMesh, withConstrain, surfaceColor, poissonRatio, youngModulus, totalMass, solver)
 
-    @staticmethod
+    #@staticmethod
```
Comment (damienmarchal) [inline python/stlib/physics/deformable/elasticmaterialobject.py]: Why

File: python/splib/geometric/gmsh.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+           :param float Mesh_CharacteristicLengthFactor:
+           :param float Mesh_CharacteristicLengthMax:
+           :param float Mesh_CharacteristicLengthMin: 
+           :param float View_GeneralizedRaiseZ
+        """ 
+        import gmshpy        
+        import os
+        import numpy as np
+        import locale
+        locale.setlocale(locale.LC_ALL, 'C')
```
Comment (damienmarchal) [inline python/splib/geometric/gmsh.py]: if this is not already done... use a contextmanager here to restore the initial locale.

Comment (damienmarchal) [reply python/splib/geometric/gmsh.py]: with mycontextManager.locale("C"):
       ...

File: python/splib/geometric/gmsh.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        return OutputFilePath
+
+def createScene(root):
+        from stlib.scene import Scene
+
+        Scene(root)
+        root.VisualStyle.displayFlags="showForceFields"
+
+        # The list of mesh (e.g. Mesh_CharacteristicLengthFactor), geometry, view, etc. options can be found here: http://gmsh.info/doc/texinfo/gmsh.html, Appendix B
+        filename = meshFromParametricGeometry(filepath='/home/stefan/Downloads/Cube.stl',
```
Comment (damienmarchal) [inline python/splib/geometric/gmsh.py]: I disagree with the name ...

**Comment by damienmarchal** [comment]
Well, you can also update the documentation so it work with readthedocs

https://stlib.readthedocs.io/en/latest/

**Comment by sescaida** [comment]
Addressed all the issues! PR is ready for review!

File: python/stlib/physics/deformable/elasticmaterialobject.py [hunk 4]
```diff
@@ -69,7 +69,7 @@ def createPrefab(self,
             self.solver = self.node.createObject('SparseLDLSolver', name="solver")
 
         self.container = self.node.createObject('TetrahedronSetTopologyContainer', src='@loader', name='container')
-        self.dofs = self.node.createObject('MechanicalObject', template='Vec3d', name='dofs')
+        self.dofs = self.node.createObject('MechanicalObject', template='Vec3d', name='tetras')
```
Comment (damienmarchal) [inline python/stlib/physics/deformable/elasticmaterialobject.py]: ???? probably a mistake...please remove that. Otherwise looks good to me.

Comment (sescaida) [reply python/stlib/physics/deformable/elasticmaterialobject.py]: ok, corrected!

**Comment by EulalieCoevoet** [comment]
If it looks good for everyone, let's merge this PR :)
Thanks @sescaida

**Pull request merged**